warden/x/act/keeper/actions_keeper.go

```diff
@@ -99,3 +109,75 @@ func (k ActionKeeper) Import(ctx sdk.Context, actions []types.Action) error {
 func (k ActionKeeper) Coll() repo.SeqCollection[types.Action] {
 	return k.actions
 }
+
+func (k ActionKeeper) pruneAction(ctx context.Context, action types.Action, blockHeight int64) error {
+	if err := k.actions.Remove(ctx, action.Id); err != nil {
+		return err
+	}
+
+	for _, addr := range action.Mentions {
+		key := collections.Join(sdk.MustAccAddressFromBech32(addr), action.Id)
+		if err := k.actionByAddress.Remove(ctx, key); err != nil {
+			return err
+		}
+	}
+
+	if err := k.previousPruneBlockHeight.Set(ctx, blockHeight); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (k ActionKeeper) GetLatestPruneHeight(ctx context.Context) (int64, error) {
+	hasItem, err := k.previousPruneBlockHeight.Has(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	if !hasItem {
+		return 0, nil
+	}
+
+	return k.previousPruneBlockHeight.Get(ctx)
```

Comment:
sorry i haven't caught this before, it's a small optimization but I think we can do in one db call instead of two, like this:

```suggestion
    h, err := k.previousPruneBlockHeight.Get(ctx)
    if errors.Is(err, collections.ErrNotFound) {
        return 0, nil
    }
    return h, err
```